src/ui/inventory.js

```diff
@@ -0,0 +1,182 @@
+import { CONSTANTS, FONTS } from "../constants/constants.js";
+import { getItemClass } from "./itemTypes.js";
+import { ITEMS } from "../constants/items.js";
+import { MATERIALS } from "../constants/materials.js";
+
+export class Inventory {
+    scene;
+    menu;
+    playerItems = []; // Pointer to cookies, store item name/material
+    inventory = []; // Images
+    curSelectedItemIndex = -1;
+
+    constructor(scene, inventory) {
+        this.scene = scene;
+        this.playerItems = inventory;
+
+        // Update and show inventory on startup
+        this.playerItems.forEach(item => {
+            // Add blanks
+            if (!Object.keys(item).length) {
+                this.addToInventory({});
+            }
+            // Add items
+            else {
+                let itemObj = getItemClass(item.name, item.material, this.scene);
+                this.addToInventory(itemObj);
+            }
+        });
+    }
+
+    async addToInventory(item) {
+        let index = await this.checkForEmptySlot();
+
+        // Add items
+        if (Object.keys(item).length && (this.inventory.length < 28 || index >= 0)) {
+            // Repopulate on startup
+            if (this.playerItems.length > this.inventory.length) {
+                index = this.inventory.length;
+            }
+            // Add items into first empty slot
+            else if (index >= 0) {
+                this.playerItems[index] = {
+                    name: item.item, 
+                    material: item.material
+                };
+            }
+            // Add new item to end
+            else if (this.playerItems.length == this.inventory.length) {
+                this.playerItems.push({
+                    name: item.item, 
+                    material: item.material
+                }); 
+                index = this.inventory.length;
+            }
+
+            // Add item images
+            let column = index % 4;
+            let row = Math.floor(index / 4);
+            let x = 570 + column * 45;
+            let y = 225 + row * 35;
+            item.addToInventory(x, y, index);
+
+            // Hide if inventory is not selected
+            if (this.scene.currentPanel != CONSTANTS.PANEL.INVENTORY) {
+                item.show(false);
+            }
+
+            this.inventory[index] = item;
+        } else if (this.inventory.length < 28) {
+            this.inventory.push({});
+        } else {
+            console.log("Inventory is full");
+        }
+    }
+
+    async checkForEmptySlot() {
+        for (let index = 0; index < this.playerItems.length; index++) {
+            if (!Object.keys(this.playerItems[index]).length) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    createRightClickMenu(x, y, item, index) {
```

Comment:
We will probably want the background image for the right click be automatically sized based on the number of options

Reply:
Yeah I'm trying to think of how to do this since I'm using a screenshot of the in-game right click menu. 
1) Take multiple screenshots of different sizes and load right-click-[#options]
2) Draw the menu ourselves using the Phaser.Graphics object

Reply:
Multiple screenshots seems like an ape-like thing to do but honestly whichever looks better works for me lol

Reply:
I can't think of any circumstance where you would need more than 4 so I think it might be the easiest and potentially the fastest way assuming the images are loaded on startup rather than dynamically. If you think there would be more than that i think would definitely look into how to draw it. For now I'll just add it as a task, not a priority rn imo

Reply:
If we have degradable items there would be a "check status" or w/e. Also in a bank or shop interface there could be a lot of options when right clicking.

Reply:
Yeah I think we should have a separate item class specifically for shop items because all shop items have the same buy options that aren't related to the items at all outside of their examine text.